controllers/storagecluster/placement.go

```diff
@@ -20,6 +20,11 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookCephv1.Placeme
 	} else {
 		in := defaults.DaemonPlacements[component]
 		(&in).DeepCopyInto(&placement)
+		// default value for mds pod anti affinity is set to the filesystem name
+		// label rook_file_system is added to the mds pod using rook operator
+		if component == "mds" && placement.PodAntiAffinity != nil {
```

Comment:
Can we just directly assign the antiaffinity here? It seems so awkward to add a placeholder that will be updated later. This way we just directly generate it when needed. 
```suggestion
		if component == "mds" {
			placement.PodAntiAffinity = getMdsWeightedPodAffinityTerm(100, generateNameForCephFilesystem(sc))
		}
```

Reply:
done